### DISCUSSION TITLE
Cell is not calculated

### DISCUSSION THREAD

**Discussion by ValMati** [ClosedXML/ClosedXML#2113] [Q&A]
Hi, I have an application that using ClosedXML gives value to different cells to get the calculated value in other cells. Everything works perfectly on several computers except for one specific computer. On this computer there is a specific Excel file in which one of its cells is not calculated.

This cell has a formula by which two other cells containing decimal numerical values are multiplied. At the first place I thought that the origin of the problem was the regional configuration, but the configuration of that computer is the same as the rest and also in that same Excel file other operations with decimals are done without any problem.

I force the recalculation with "XLWorkbook.RecalculateAllFormulas()" and everything is recalculated, except one of the cells.

NuGet version: 0.101.0

Regards and thanks

**Comment by jahav** [comment]
Try to invalidate just the cell or it's precedents through `IXLCell.InvalidateFormula()` and check `IXLCell.NeedsRecalculation` flag. The whole modification detection buggy and slated for replacement. 

Culture would have effect only if there was string to number conversion (check if original decimal numerical values are not in fact strings that only look like numbers through `IXLCell.Value.Type`) and even then it would return some `XLError` on impossible. Formula should always calculate *something*, trouble with number `XLError` or empty string.  Note that `IXLCell.Value` try to evaluate formula, if necessary, but `IXLCell.CachedValue` just returns the current (possibly obsolete value)

**Reply by ValMati** [reply]
First of all, thank you very much for your reply.

The cells that I have to assign a value to I do it by calling _SetValue()_ passing it a double, so it doesn't seem to be the problem.

The NeedsRecalculation value of the problematic cell is always false, before and after changing the value of the cells on which it depends. If I call _InvalidateFormula()_ and then start assigning values to the other cells then it does calculate it correctly. Is there any way to invalidate all cells with formula?

**Reply by ValMati** [reply]
I tried with _RecalculateAllFormulas()_ and it didn't work :(
